### PR TITLE
Update SRML Council: prevent excess votes/computation

### PR DESCRIPTION
**Pull request by ltfschoen**
* Add assertion and associated tests to prevent a vote from voters that provide a list of votes with a length that exceeds the amount of candidates, since otherwise an attacker may be able to submit a very long list of `votes` that far exceeds the amount of candidates and waste more computation than a reasonable voting bond would cover. Added additional associated test that may be run with `cargo test -p srml-council`

* Add assertion and associated tests so that amount of candidates must be greater than zero when calling `set_approvals` for a list of votes to be made by a voter 

* Update comment to refer to `reporter` instead of `who` (target of inactivity), since the origin is the `reporter`

* Update comment to refer more specifically to how many vote indexes, since `InactiveGracePeriod` is measured in vote indexes

* Update comment for `ApprovalsOf` since previously the comment was a duplicate of the comment for `LastActiveOf`'s

* Create variable to refer to `retaining_seats` to improve readability

* Reference Notes: https://hackmd.io/nr6kPD2sR4urmljtvHs0CQ

### CODE REVIEW

File: srml/council/src/seats.rs [hunk 1]
```diff
@@ -94,9 +94,25 @@ decl_module! {
...
+			let expiring = active_council.iter().take_while(|i| i.1 == number).map(|i| i.0.clone()).collect::<Vec<_>>();
+			let retaining_seats = active_council.len() - expiring.len();
+			let empty_seats = desired_seats - retaining_seats;
 
 			ensure!(!Self::presentation_active(), "no approval changes during presentation period");
 			ensure!(index == Self::vote_index(), "incorrect vote index");
+			ensure!(candidates.len() > 0, "amount of candidates to receive approval votes must be greater than zero");
+			// Prevent a vote from voters that provide a list of votes that exceeds the candidates length
+			// since otherise an attacker may be able to submit a very long list of `votes` that far exceeds
+			// the amount of candidates and waste more computation than a reasonable voting bond would cover.
```
[the remark this answers is not in the document]
Comment (ltfschoen) [reply srml/council/src/seats.rs]: would you like me to attempt the second step? if so would you prefer i update this PR or propose the update in a separate PR?

**Comment by ltfschoen** [comment]
i've pushed commit https://github.com/paritytech/substrate/pull/1162/commits/db7b90e7f688fe73963ad0953ed43cb69938ea5b to address relevant comment that arose in a different PR https://github.com/paritytech/substrate/pull/1185#discussion_r238141545

File: srml/council/src/seats.rs [hunk 2]
```diff
@@ -344,13 +352,15 @@ decl_storage! {
 
 		// permanent state (always relevant, changes only at the finalisation of voting)
 		/// The current council. When there's a vote going on, this should still be used for executive
-		/// matters.
+		/// matters. The block number (second element in the tuple) is the block that their position is
+		/// active until (calculated by the sum of the block number when the council member was elected
+		/// and their term duration.
```
Comment (gavofyork) [inline srml/council/src/seats.rs]: missing `)`
